include/openenclave/internal/cpuid.h

```diff
@@ -7,7 +7,7 @@
 #include <openenclave/bits/defs.h>
 
 #define OE_CPUID_OPCODE 0xA20F
-#define OE_CPUID_LEAF_COUNT 8
+#define OE_CPUID_LEAF_COUNT 17 /* 0-7, 0x80000000-0x800000008*/
 #define OE_CPUID_EXTENDED_CPUID_LEAF 0x80000000
```

Comment:
Remove this definition if it is no longer used (it's only reference seems to be in the test for sigill_handling.c)

Reply:
Unrelated question (ok to handle outside of this PR): is cpuid.h only for SGX? assuming so, seems like the file needs a rename or be moved under an sgx subdirectory.

Reply:
The unused definition has been removed. 
As to the cpuid.h,  based on current OE implementation, it seems the cpuid collection and usage are indeed for SGX only.  Not sure about your plan for OP-TEE,  so I keep the file as it is in the latest commit.

Reply:
Sounds like the simplest change (ok in a separate PR or in this PR, I have no preference) is to move cpuid.h down a level under internal/sgx